projects/hathor-wallet-mobile/0002-nano-contract-integration.md

```diff
@@ -0,0 +1,476 @@
+
+The modal presents detailed information containing *Blueprint ID*, *Blueprint Method*, *Caller*, *Nano Contract ID* (if applicable), *Action List*, and *Argument List*. An *Action List* can be either types *deposit* or *withdrawal*. Both the lists can be empty.
+
+##### Design Bill
+- **Placement**:
+	- The modal must render centrally on top of the current screen, ensuring clear visibility and focus.
+	
+- **Functionality**:
+	- Include detailed transaction information for user review:
+		- *Blueprint ID*, *Blueprint Method*, *Caller*, *Nano Contract ID* (if applicable), *Action List*, and *Argument List*.
+	- Include detailed information for each *Action List* item:
+		- *Type*, *Token* and *Data*.
+	- Include detailed information for *Data*:
+		- *Amount*, *Address*
+	- Include detailed information for each *Argument List* item:
+		- *Type* and *Value*
+	- Provide clear options for user approval or rejection of the transaction.
+	- Upon tapping *Approve*, initiate the processing with appropriate loading feedback.
+
+- **Visibility Conditions**:
+	- Ensure the modal is visible whenever a *Nano Contract Transaction Request* is made.
+	- Make sure to always display *Action List* and *Argument List* when they are empty.
+	- In case of processing success, display a success feedback.
+	- Must not display the *Nano Contract ID* when the *Blueprint Method* value is `initialize`.
+	- Differentiate item *Action List* item types *deposit* and *withdrawal* using distict visual indicators.
+
+- **Error Management**:
+	- In case of processing error, display an error feedback.
+
+>[!NOTE]
+>The *Caller* represents the wallet address calling the Blueprint method.
+
+>[!TIP]
+>Use the following value examples:
+>- *Blueprint ID*
+>	- eg: `494d0ac59a6918b771b122395206fef3f349f84f20dc430188a319d4ead24a3b`
+>- *Blueprint Method*
+>	- eg: `initiale`, `swap`, `bet`
+>- *Caller*
+>	- eg: `WRqBSi5jc74P8tRo7CmgiGByA1eTzfMh7B`
+>- *Nano Contract ID*
+>	- eg: `000001e43dda7348c53c6252d98799c9e2a0b4ea3b5f10b1182578d643c9641d`
+>- *Action List* item:
+>	- *Type*
+>		- eg: `deposit`, `withdrawal`
+>	- *Token*
+>		- eg: `000001a176aa72b84f3486b3fcea327b023a70d1b6461c1b6a9073785b5775d8`
+>	- *Data*
+>		- *Amount*
+>			- eg: `1.23`
+>		- *Address*
+>			- eg: `WZhKusv57pvzotZrf4s7yt7P7PXEqyFTHk`
+>- *Argument List* item:
+>	- *Type*
+>		- eg: `byte`, `int`, `string`, `boolean`
+>	- *Value*
+>		- eg: `000002d068a46c1c0a6ae84ca3b3c1984d517c8a3783c2ceb024cd5355595851` for byte
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+## Load Nano Contract History
+
+##### When execute
+- It should happen everytime a new *Nano Contract Transaction* containing a caller from user's wallet is processed on `sagas/wallet.js@handleTx`.
```

Comment:
Should we also reload the history when a nano contract which the user has registered state mutated?

Reply:
Well, the only way to mutate nano contract's state is by a transaction. Therefore, this is already capturing the state mutation.

But when you say "reload the history" are you referring to a re-render of the screen?

Reply:
> Well, the only way to mutate nano contract's state is by a transaction. Therefore, this is already capturing the state mutation.

Oh, ok!